analyzers/src/SonarAnalyzer.CSharp/Extensions/AssignmentExpressionSyntaxExtensions.cs

```diff
@@ -0,0 +1,113 @@
+﻿/*
+ * SonarAnalyzer for .NET
+ * Copyright (C) 2015-2022 SonarSource SA
+ * mailto: contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System.Collections.Immutable;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using StyleCop.Analyzers.Lightup;
+
+namespace SonarAnalyzer.Extensions
+{
+    internal static class AssignmentExpressionSyntaxExtensions
+    {
+        public readonly record struct AssignmentMapping(CSharpSyntaxNode Left, ExpressionSyntax Right);
+
+        /// <summary>
+        /// Maps the left and the right side arguments of an <paramref name="assignment"/>. If both sides are tuples, the tuple elements are mapped.
+        /// <code>
+        /// (var x, var y) = (1, 2);                 // [x→1, y→2]
+        /// (var x, (var y, var z)) = (1, (2, 3));   // [x→1, y→2, z→3]
+        /// var x = 1;                               // [x→1]
+        /// (var x, var y) = M();                    // [(x,y)→M()]
+        /// </code>
+        /// </summary>
+        /// <param name="assignment">The <paramref name="assignment"/> expression.</param>
+        /// <returns>A mapping from expressions on the left side of the <paramref name="assignment"/> to the right side.</returns>
+        public static ImmutableArray<AssignmentMapping> MapAssignmentArguments(this AssignmentExpressionSyntax assignment)
+        {
+            // (var x, var y) = (1, 2)
+            if (TupleExpressionSyntaxWrapper.IsInstance(assignment.Left)
+                && TupleExpressionSyntaxWrapper.IsInstance(assignment.Right))
+            {
+                var left = (TupleExpressionSyntaxWrapper)assignment.Left;
+                var right = (TupleExpressionSyntaxWrapper)assignment.Right;
+                var builder = ImmutableArray.CreateBuilder<AssignmentMapping>(initialCapacity: left.Arguments.Count);
+                MapTupleElements(builder, left, right);
+                return builder.ToImmutableArray();
+            }
+            // var (x, y) = (1, 2)
+            else if (DeclarationExpressionSyntaxWrapper.IsInstance(assignment.Left)
+                && (DeclarationExpressionSyntaxWrapper)assignment.Left is { Designation: { } leftDesignation }
```

Comment:
I think it would be better to remove the named arguments as they are creating some noise.
I noticed this is not explicit in our [coding style](https://github.com/SonarSource/sonar-dotnet/blob/master/docs/coding-style.md) guide; I added a line to discuss in the next "How we work meeting".

PS: Consider this comment valid for all the lines where there's a named argument.

Reply:
These are pattern matches and not named arguments. It is roughly equivalent to
```cs
var a = (DeclarationExpressionSyntaxWrapper)assignment.Left;
if (a.Designation is not null)
{
  var leftDesignation  = a.Designation;
}
```

Reply:
@martin-strecker-sonarsource  🤦 Sorry for the noise - out of velocity I picked the wrong line.
There are some named arguments in other spots though. Could you please remove them?
Should be [here ](https://github.com/SonarSource/sonar-dotnet/blob/ed765f05b3f02e3f6d5c65b14e5c5df6025c77d2/analyzers/src/SonarAnalyzer.CSharp/Extensions/AssignmentExpressionSyntaxExtensions.cs#L51) and [here](https://github.com/SonarSource/sonar-dotnet/blob/ed765f05b3f02e3f6d5c65b14e5c5df6025c77d2/analyzers/src/SonarAnalyzer.CSharp/Extensions/AssignmentExpressionSyntaxExtensions.cs#L63).

Reply:
>  I added a line to discuss in the next "How we work meeting"

I already did that a few weeks ago, but we never did it to that point. But thanks for adding it anyway.